### DISCUSSION TITLE
Generic Panel Extension Support

### DISCUSSION THREAD

**Discussion by valenotary** [zed-industries/zed#18880] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

(I could've sworn there was some issue/PR for this already but I couldn't find it so I do apologize for asking again if it was already there and I missed it)

I want to build my own editor on top of Zed, but I want to be able to package it as an installable extension (and not just as a fork of the project). I need the ability to create custom panes, tabs... similar to the API that the IntelliJ editors provide (but in Rust, not in Java :)).

- Is this already on the roadmap?
  - If so, ETA?
  - If not, why not?

### If applicable, add mockups / screenshots to help present your vision of the feature

TODO

**Comment by SomeoneToIgnore** [comment]
Here are the current extension capabilities: https://zed.dev/docs/extensions/developing-extensions
Also: https://github.com/zed-industries/zed/blob/cf89604fb5ca8880824e0d7e25b049b8c45c1ece/crates/extension_api/src/extension_api.rs#L60-L150

There's no plans for the near future to extend the extension API in the direction of the visual components, as we have to deal with the very basics before doing any advanced stuff + there's tons of other things required by people, not only the advanced extensions API.
Some basics like text editing are being worked on here: https://github.com/zed-industries/zed/pull/17809
